src/Tribe/API.php

```diff
@@ -602,5 +642,222 @@ public static function get_and_flatten_event_meta( $event_id ) {
 
 			return $post_meta;
 		}
+
+		/**
+		 * Sanitize the arguments array before sending to create/update an event post.
+		 *
+		 * Use this prior to sending arguments to post create/update function.
+		 *
+		 * @since TBD
+		 *
+		 * @see Tribe__Events__API::sanitize_meridian_meta_value
+		 * @see Tribe__Events__API::sanitize_hour_meta_value
+		 * @see Tribe__Events__API::sanitize_minute_meta_value
+		 *
+		 * @param array $args The arguments sent to create/update an event post.
+		 *
+		 * @return array|WP_Error
+		 */
+		public static function sanitize_event_post_create_update_args( $args ) {
+			if (
+				empty( $args['post_type'] )
+				|| Tribe__Events__Main::POSTTYPE !== $args['post_type']
+				|| ! is_array( $args )
```

Comment:
I will move this one first instead as before accesing the key `post_type` you need to make sure `$args` is an array.